satpy/scene.py

```diff
@@ -1081,12 +1081,106 @@ def to_xarray_dataset(self, datasets=None):
+        header_attrs:
+            Global attributes of the output xr.Dataset.
+        epoch (str):
+            Reference time for encoding the time coordinates (if available).
+            Example format: "seconds since 1970-01-01 00:00:00".
+            If None, the default reference time is retrieved using "from satpy.cf_writer import EPOCH"
+        flatten_attrs (bool):
+            If True, flatten dict-type attributes.
+        exclude_attrs (list):
+            List of xr.DataArray attribute names to be excluded.
+        include_lonlats (bool):
+            If True, it includes 'latitude' and 'longitude' coordinates.
+            If the 'area' attribute is a SwathDefinition, it always includes
+            latitude and longitude coordinates.
+        pretty (bool):
+            Don't modify coordinate names, if possible. Makes the file prettier,
+            but possibly less consistent.
+        include_orig_name (bool).
+            Include the original dataset name as a variable attribute in the xr.Dataset.
+        numeric_name_prefix (str):
+            Prefix to add the each variable with name starting with a digit.
+            Use '' or None to leave this out.
+
+        Returns
+        -------
+        ds, xr.Dataset
+            A CF-compliant xr.Dataset
+
+        """
+        from satpy.writers.cf_writer import EPOCH, collect_cf_datasets
+
+        # Retrieve epoch
+        if epoch is None:
+            epoch = EPOCH
+
+        # Get list of DataArrays
+        # - If datasets=None, retrieve all loaded datasets
+        if datasets is None:
+            datasets = list(self.keys())  # list DataIDs
+        list_dataarrays = self._get_dataarrays_from_identifiers(datasets)
+
+        # Check that some DataArray could be returned
+        if len(list_dataarrays) == 0:
+            return xr.Dataset()
+
+        # Collect xr.Dataset for each group
+        grouped_datasets, header_attrs = collect_cf_datasets(list_dataarrays=list_dataarrays,
+                                                             header_attrs=header_attrs,
+                                                             exclude_attrs=exclude_attrs,
+                                                             flatten_attrs=flatten_attrs,
+                                                             pretty=pretty,
+                                                             include_lonlats=include_lonlats,
+                                                             epoch=epoch,
+                                                             include_orig_name=include_orig_name,
+                                                             numeric_name_prefix=numeric_name_prefix,
+                                                             groups=None)
+        if len(grouped_datasets) == 1:
+            ds = grouped_datasets[None]
+            return ds
+        else:
+            msg = """The Scene object contains datasets with different areas.
```

Comment:
codecov says this is not tested?

Reply:
@mraspaud can't be tested because if more than 1 area, `collect_cf_datasets` raise an error (and groups is an argument of  `to_xarray`. What should I do?